front/src/applications/rollingStockEditor/components/RollingStockEditorCurves.tsx

```diff
@@ -1,36 +1,130 @@
 import RollingStock2Img from 'common/RollingStockSelector/RollingStock2Img';
 import { listCurvesComfort } from 'common/RollingStockSelector/RollingStockCardDetail';
 import RollingStockCurve from 'common/RollingStockSelector/RollingStockCurves';
-import { RollingStock } from 'common/api/osrdEditoastApi';
+import { Comfort, RollingStock } from 'common/api/osrdEditoastApi';
 import { isEmpty } from 'lodash';
-import React, { Dispatch, SetStateAction, useState } from 'react';
+import React, { Dispatch, SetStateAction, useEffect, useState, useCallback } from 'react';
 import { useTranslation } from 'react-i18next';
 import Spreadsheet, { CellBase, Matrix, createEmptyMatrix } from 'react-spreadsheet';
 import { emptyStringRegex } from 'utils/strings';
+import {
+  updateComfortLvl,
+  updateElectricalProfile,
+  updateTractionMode,
+} from 'reducers/rollingstockEditor';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  getComfortLevel,
+  getElectricalProfile,
+  getTractionMode,
+} from 'reducers/rollingstockEditor/selectors';
+import SelectorSNCF from '../../../common/SelectorSNCF';
+import { SelectedCurve, Curve, RollingStockSelectorParams } from '../consts';
 
-type Curve = {
-  max_efforts: number[];
-  speeds: number[];
-};
-
-export default function RollingStockEditorCurves(props: {
+export default function RollingStockEditorCurves({
+  data,
+  currentRsEffortCurve,
+  setCurrentRsEffortCurve,
+}: {
   data?: RollingStock;
   currentRsEffortCurve: RollingStock['effort_curves'];
   setCurrentRsEffortCurve: Dispatch<SetStateAction<RollingStock['effort_curves']>>;
 }) {
-  const { data, currentRsEffortCurve, setCurrentRsEffortCurve } = props;
-
   const { t } = useTranslation('rollingstock');
-  const selectedMode = currentRsEffortCurve ? Object.keys(currentRsEffortCurve.modes)[0] : ''; // 1er mode par défaut.
+  const dispatch = useDispatch();
+
   const EMPTY_MATRIX = createEmptyMatrix<CellBase<string>>(8, 2);
 
-  // TODO: recup les listes avec filtres pour multi courbes
+  const selectedElectricalProfile = useSelector(getElectricalProfile);
+  const selectedMode = useSelector(getTractionMode);
```

Comment:
If 'mode' and 'selectedMode' correspond to tractionMode, it is very important to name that as such, otherwise it is hard to read